File: 746_Min_Cost_Climbing_Stairs/CPP/Tabulation/Create 746_Min_Cost_Climbing_Stairs_tabu.cpp

class Solution {
public:

int minCostClimbingStairs(vector<int>& cost) {
    
    int dpSize = cost.size()+2;
    int dp[dpSize];
    dp[dpSize-1] = dp[dpSize-2] = 0;
    
    for(int i=dpSize-3;i>=0;i--)
    {
        dp[i] = cost[i] + min(dp[i+1],dp[i+2]);
    }
    
    return min(dp[0],dp[1]);
    
}

};

File: 746_Min_Cost_Climbing_Stairs/Java/Tabulation/Create 746_Min_Cost_Climbing_Stairs_tabu.java
class Solution {
    
    public int minCostClimbingStairs(int[] cost) {
        
        int dpSize = cost.length+2;
        int dp[] = new int[dpSize];
        Arrays.fill(dp,-1);
        
        
        dp[dpSize-1] = dp[dpSize-2] = 0;
        
        for(int i=dpSize-3;i>=0;i--)
        {
            dp[i] = cost[i] + Math.min(dp[i+1],dp[i+2]);
        }
        return Math.min(dp[0],dp[1]);
    }
    
}
